src/components/SlateEditor/plugins/embed/SlateAudio.jsx

```diff
@@ -21,55 +22,92 @@ class SlateAudio extends React.Component {
     super(props);
     this.state = { audio: {} };
     this.toggleEdit = this.toggleEdit.bind(this);
+    this.onAudioFigureInputChange = this.onAudioFigureInputChange.bind(this);
   }
 
   async componentWillMount() {
     try {
       const audio = await visualElementApi.fetchAudio(
         this.props.embed.resource_id,
       );
-      this.setState({ audio: { ...audio, title: audio.title.title } });
+      this.setState({
+        audio: {
+          ...audio,
+          title: audio.title.title,
+          caption: this.props.embed.caption,
+        },
+      });
     } catch (error) {
       visualElementApi.onError(error);
     }
   }
 
+  onAudioFigureInputChange(e) {
+    const { value, name } = e.target;
+
+    this.setState(prevState => ({
+      audio: {
+        ...prevState.audio,
+        [name]: value,
+      },
+    }));
+    this.props.onFigureInputChange(e);
+  }
+
   toggleEdit(e) {
     e.stopPropagation();
     this.setState(prevState => ({ editMode: !prevState.editMode }));
   }
 
   render() {
-    const { attributes, onFigureInputChange, embed, ...rest } = this.props;
-    const { audio = {} } = this.state;
+    const {
+      attributes,
+      onFigureInputChange,
+      submitted,
+      embed,
+      t,
+      ...rest
+    } = this.props;
+    const { audio } = this.state;
 
     const player = audio.id && (
+      <AudioPlayerMounter audio={audio} speech={embed.audioType === 'speech'} />
+    );
+
+    return (
       <Figure id={`${audio.id}`} {...attributes}>
-        <AudioPlayerMounter
-          audio={audio}
-          speech={embed.audioType === 'speech'}
+        {this.state.editMode ? (
+          <EditAudio
+            onExit={() => this.setState({ editMode: false })}
+            audioType={embed.audioType || 'sound'}
+            onChange={onFigureInputChange}
+            embed={embed}
+            t={t}
+            {...rest}>
+            {player}
+          </EditAudio>
+        ) : (
+          <div
```

Comment:
Hvorfor ikke en faktisk button?

Reply:
En `Button` vil kreve ytterligere CSS-endringer for "nullstille" knappstylen. Siden det ikke er en knapp men egentlig en slags overlay, foreslår jeg at vi beholder det slik.